recap/src/lib.rs

```diff
@@ -341,31 +386,55 @@ impl<'a: 'de, 'de, Iter: Iterator<Item = (&'a str, &'a str)>> de::Deserializer<'
 fn from_iter<'a, Iter, T>(iter: Iter) -> Result<T>
 where
     T: de::Deserialize<'a>,
-    Iter: IntoIterator<Item = (&'a str, &'a str)>,
+    Iter: IntoIterator<Item = (&'a str, &'a str, Option<&'a FieldOptions>)>,
 {
     T::deserialize(Deserializer::new(iter.into_iter()))
 }
 
+/// Options for a given field
+pub struct FieldOptions {
+    /// If given for a sequence field, will split the captured string using this Regex. Otherwise
+    /// sequence fields split on comma (`","`).
+    pub delimiter_regex: Option<Regex>,
+}
+
 /// Deserialize a type from named regex capture groups
 ///
 /// See module level documentation for examples
 pub fn from_captures<'a, D>(
     re: &'a Regex,
     input: &'a str,
 ) -> Result<D>
+where
+    D: Deserialize<'a>,
+{
+    from_captures_with_options(re, input, None)
+}
+
+/// Deserialize a type from named regex capture groups. Allows specifying options for each field.
+///
+/// See module level documentation for examples
+pub fn from_captures_with_options<'a, D>(
+    re: &'a Regex,
+    input: &'a str,
+    field_options: Option<&'a HashMap<String, FieldOptions>>,
+) -> Result<D>
 where
     D: Deserialize<'a>,
 {
     let caps = re.captures(input).ok_or_else(|| {
         envy::Error::Custom(format!("No captures resolved in string '{}'", input))
     })?;
-    from_iter(
-        re.capture_names()
-            .map(|maybe_name| {
-                maybe_name.and_then(|name| caps.name(name).map(|val| (name, val.as_str())))
-            })
-            .flatten(),
-    )
+
+    from_iter(re.capture_names().flatten().filter_map(|name| {
+        caps.name(name).map(|val| {
+            (
+                name,
+                val.as_str(),
+                field_options.and_then(|fo| fo.get(name)),
+            )
+        })
+    }))
```

Comment:
Some of the reworking of chained methods here also came from clippy's suggestions.